research-software-directory/RSD-as-a-service
# feat: use OpenAlex for all unsupported DOI registry agents

## Use OpenAlex for unsupported DOI RAs

### Changes proposed in this pull request

* On the frontend, when searching for a DOI that is not from Crossref or DataCite, use OpenAlex instead
* Same when bulk importing on the frontend
* Same for the mentions scraper

### How to test

* `docker compose down --volumes && docker compose build --parallel && docker compose up --scale data-generation=0`
* Now that there are no mentions, create a software or project page, search for `10.3968/10006`, it should be found using OpenAlex, but don't add it
* Do a bulk import of mentions (use e.g. #808) including this DOI, it should be found, import them
* Run the mentions scraper (a few times if necessary):
* `docker compose exec scrapers java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.doi.MainMentions`
* The `scraped_at` field at http://localhost/api/v1/mention should be set, no errors should be generated
* Do some more testing with other DOIs (use e.g. #808) to see if everything still works


Closes #1321

PR Checklist:
